CMakeLists.txt

```diff
@@ -26,9 +26,18 @@ if(env_cxx_standard GREATER -1)
       WARNING "C++ standard version definition detected in environment variable."
       "PyTorch requires -std=c++14. Please remove -std=c++ settings in your environment.")
 endif()
+
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_C_STANDARD 11)
 
+# https://developercommunity.visualstudio.com/t/VS-16100-isnt-compatible-with-CUDA-11/1433342
+if(MSVC)
+  if(USE_CUDA)
+    set(CMAKE_CXX_STANDARD 17)
+  endif()
```

Comment:
But that wouldn't work when TorchAudio is compiled with CUDA-10.2, would it?

Reply:
It would.
https://app.circleci.com/pipelines/github/pytorch/audio/7396/workflows/058b3c2c-8421-40c2-915c-a8441deac091/jobs/339245.
It's a Visual Studio Issue. I verified that another workaround is to downgrade the Visual Studio to 16.8.5 which PyTorch Builder is using.
But I think user experience might better that the user doesn't need to select the compiler with the specific version.

Reply:
The root cause is a regression in VS. https://github.com/microsoft/STL/issues/2020